 (FROM MASTER) init: fix writepid wrong pid written in file(s)

When child (pid==0) executes fork branch 'pid_str'
will always be 0 writting wrong value to file(s).
Fix by utilizing getpid() to obtain correct pid.

Tracker link provides some insight on why use cases
aren't broken.

Bug: https://code.google.com/p/android/issues/detail?id=200392

diff --git a/init/init.cpp b/init/init.cpp
index 030d208..13fc5fa 100644
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -305,7 +305,7 @@ void service_start(struct service *svc, const char *dynamic_args)
         scon = NULL;
 
         if (svc->writepid_files_) {
-            std::string pid_str = android::base::StringPrintf("%d", pid);
+            std::string pid_str = android::base::StringPrintf("%d", getpid());
             for (auto& file : *svc->writepid_files_) {
                 if (!android::base::WriteStringToFile(pid_str, file)) {
                     ERROR("couldn't write %s to %s: %s\n",
